presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1517,11 +1520,30 @@ else if (expression instanceof LongLiteral) {
                     }
 
                     // otherwise, just use the expression as is
+                    boolean checkDeprecatedBehavior = false;
                     if (orderByExpression == null) {
                         orderByExpression = expression;
+                        checkDeprecatedBehavior = true;
                     }
 
                     ExpressionAnalysis expressionAnalysis = analyzeExpression(orderByExpression, sourceScope);
+                    if (checkDeprecatedBehavior) {
+                        for (Map.Entry<NodeRef<Expression>, FieldId> columnExpression : expressionAnalysis.getColumnReferences().entrySet()) {
+                            Optional<ResolvedField> resolvedField = orderByScope.tryResolveField(columnExpression.getKey().getNode());
+                            if (resolvedField.isPresent() &&
+                                    !FieldId.from(resolvedField.get()).equals(columnExpression.getValue())) {
+                                warningCollector.add(new PrestoWarning(
+                                        WarningCode.LEGACY_ORDER_BY,
+                                        format("Column %s referenced in ORDER BY expressions is present " +
```

Comment:
`The column %s referenced in the ORDER BY expression is also present in the SELECT clause aliases and the source table.`